core/services/workflows/v2/secrets.go

```diff
@@ -75,12 +79,37 @@ func (s *secretsFetcher) GetSecrets(ctx context.Context, request *sdkpb.GetSecre
 	return resp, err
 }
 
-func (s *secretsFetcher) getSecrets(ctx context.Context, request *sdkpb.GetSecretsRequest) ([]*sdkpb.SecretResponse, error) {
+func (s *secretsFetcher) getSecretsForBatch(ctx context.Context, request *sdkpb.GetSecretsRequest) ([]*sdkpb.SecretResponse, error) {
 	vaultCap, err := s.capRegistry.GetExecutable(ctx, vault.CapabilityID)
 	if err != nil {
 		return nil, errors.New("failed to get vault capability: " + err.Error())
 	}
 
+	localNode, err := s.capRegistry.LocalNode(ctx)
+	if err != nil {
+		return nil, errors.New("failed to get local node: " + err.Error())
+	}
+
+	vaultCapConfig, err := s.capRegistry.ConfigForCapability(ctx, vault.CapabilityID, localNode.WorkflowDON.ID)
+	if err != nil {
+		return nil, errors.New("failed to get vault capability config: " + err.Error())
+	}
+
+	value := vaultCapConfig.DefaultConfig.Underlying["VaultPublicKey"]
```

Comment:
@cedric-cordenier @DeividasK : I am assuming this is where the Vault Public Key will be available in the capability registry config, with this given map key.

Reply:
With @nadahalli we discussed using `masterPublicKey` -- can we align on that so he doesn't have to change his code?

Reply:
Starting on this topic on a slack thread

Reply:
https://chainlink-core.slack.com/archives/C092B1JV9AL/p1753992009120289